### PR TITLE
feat: added REST API for getting and setting NB VPP-Agent configuration

### PR DESCRIPTION
**Pull request by fgschwan**
Added ability to change NB VPP-Agent configuration by REST API.

- Use GET REST action on `<VPP-Agent IP address>:9191/configuration` to retrieve NB VPP-Agent configuration in YAML format.
- Use PUT REST action on `<VPP-Agent IP address>:9191/configuration` with configuration to update in YAML format in http body to update NB VPP-Agent configuration.
- Also resync is possible by using PUT REST action on `<VPP-Agent IP address>:9191/configuration?replace=true` (`<VPP-Agent IP address>:9191/configuration?replace` will do the same).

The structure of YAML in all cases is the same as for `agentctl config ...` commands. Basically, it can be viewed as REST API equivalent to `agentctl config get`, `agentctl config update`, `agentctl config update --replace`. However, the REST API output is always YAML and can't be formatted to alternative formats as in case of `agentctl` tool.

<del>There are some code dependencies to not yet merged PR https://github.com/ligato/vpp-agent/pull/1786, therefore this PR is based on the commits of that PR. That means that commits of both PRs are visible here.</del> (rebased on master)

### CODE REVIEW

**Comment by chrismetz09** [comment]
Confirming that Ligato REST docs will need update. I'll open PR once this is committed.

**Comment by chrismetz09** [comment]
Confirming this PR requires REST docs/OpenAPI updates?

**Comment by fgschwan** [comment]
> 
> 
> Confirming this PR requires REST docs/OpenAPI updates?

@chrismetz09 Hi, this PR adds new capabilities to REST API. I'm not aware that i missed to update some documentation somewhere. If this is the case, let me know where so i can fix it. Unless of course you want to do the documentation update (as stated in your other comment). However I would like to know which documentation i missed, just for the future to not repeat this mistake again.

**Comment by chrismetz09** [comment]
Hey @fgschwan , Filip,I have been maintaining Ligato Docs at docs.ligato.io. https://docs.ligato.io/en/latest/api/api-vpp-agent/ describes VPP agent REST APIs. To date, REST is GET configs only. Now PR appears to support PUT configs? So this section, elsewhere in plugins section, and separate OpenAPI definitions need updating. Docs format must be consistent. I can hack the PR, then you can review?

**Comment by fgschwan** [comment]
> 
> 
> Hey @fgschwan , Filip,I have been maintaining Ligato Docs at docs.ligato.io. [docs.ligato.io/en/latest/api/api-vpp-agent](https://docs.ligato.io/en/latest/api/api-vpp-agent/) describes VPP agent REST APIs. To date, REST is GET configs only. Now PR appears to support PUT configs? So this section, elsewhere in plugins section, and separate OpenAPI definitions need updating. Docs format must be consistent. I can hack the PR, then you can review?

@chrismetz09 
Just to clarify, the GET REST APIs with "dump" in URI that you describe in docs get actual VPP configuration by calling binary API/CLI API directly on VPP (in time of REST API call). This is southbound (SB)(if you are familiar with this kind of terminology). I made REST API for getting and setting the northbound(NB) configuration. That is the configuration that user of VPP-Agent wants to have configured inside VPP. When every config synchronization went ok (transaction are successful and there are no unresolved dependencies,i.e. missing interface with given name), the NB and SB are the same. However if something went wrong and all configuration can't be applied for whatever reason, the NB and SB configuration are different. That means the my new GET in REST API (NB config get) can get different configuration than existing "dump" GET REST API (SB config get).

Also i think that my last [previous PR](https://github.com/ligato/vpp-agent/pull/1786) is missing in REST API docs also.

Feel free to hack this PR with docs update. I will gladly review it.

PS: If you need some examples for NB configuration GET/PUT i can provide some (It is the yaml used for `agentctl config ...` commands)

**Review by milanlenco — APPROVED** [review]

File: plugins/restapi/handlers.go [hunk 1]
```diff
@@ -735,6 +730,169 @@ func convertToProtoV1(messages []protoV2.Message) []proto.Message {
...
+				"from one big configuration proto message", err, w, formatter)
+			return
+		}
+
+		// convert config messages to input for p.Dispatcher.PushData(...)
+		var configKVPairs []orchestrator.KeyVal
+		for _, configMessage := range configMessages {
+			// convert config message from dynamic to statically-generated proto message
+			// (this is needed for later processing of message - generated KVDescriptor adapters cast
+			// to statically-generated proto message and fail with dynamicpb.Message proto messages)
```
[the remark this answers is not in the document]
Comment (fgschwan) [reply plugins/restapi/handlers.go:840]: Fixed in [`ff22b3e`](https://github.com/ligato/vpp-agent/pull/1787/commits/ff22b3e3c191dd7016968b6d49a832881f9ba3df)

**Pull request merged**